save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/ManagementController.kt

```diff
@@ -0,0 +1,53 @@
+package com.saveourtool.save.demo.controller
+
+import com.saveourtool.save.demo.NewDemoToolRequest
+import com.saveourtool.save.demo.entity.GithubRepo
+import com.saveourtool.save.demo.entity.Snapshot
+import com.saveourtool.save.demo.entity.Tool
+import com.saveourtool.save.demo.service.GithubDownloadToolService
+import com.saveourtool.save.demo.service.GithubRepoService
+import com.saveourtool.save.demo.service.SnapshotService
+import com.saveourtool.save.demo.service.ToolService
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.toMono
+import reactor.kotlin.core.util.function.component1
+import reactor.kotlin.core.util.function.component2
+
+/**
+ * Internal controller that allows to add tools to db
+ */
+@RestController
+@RequestMapping("/demo/internal")
+class ManagementController(
+    private val toolService: ToolService,
+    private val githubRepoService: GithubRepoService,
+    private val snapshotService: SnapshotService,
+    private val githubDownloadToolService: GithubDownloadToolService,
+) {
+    /**
+     * @param newDemoToolRequest
+     * @return [Mono] of [Tool] entity
+     */
+    @PostMapping("/add-tool")
+    fun addTool(@RequestBody newDemoToolRequest: NewDemoToolRequest): Mono<Tool> = with(newDemoToolRequest) {
+        GithubRepo(organizationName, projectName).let { githubRepoService.saveIfNotPresent(it) }
+    }
+        .toMono()
```

Comment:
suggestion to change it to `Mono.fromCallable` to move the main operation to mono

Reply:
Ok, will fix it